### ISSUE TITLE
Seggregate/group routes as per use cases

### ISSUE THREAD

**Issue by aquibbaig** [zairza-cetb/bench-routes#284]
Currently, we display all the routes in UI in a single table. But what if the user wants to group certain routes w.r.t a certain parameter. For example: If I have an application called `blog`, I would like to group all the routes under it within the namespace of `blog`.
```
blog:
 - https://aquibbaig.in/
 - https://aquibbaig.in/projects/
 - https://aquibbaig.in/resume/
```

This feature needs to be provided as a part of quick input to either make a separate group or add it to an existing group. Lodash functions might help in this. As for the backend, we can either tag routes in `local-config.yml` based on the group they belong to or change the local-config as the way it works now, depending upon what would be suitable for the use case.

**Comment by aquibbaig** [comment]
## Proposal
----

### Frontend
- Modification of the quick input page to include a `group` entry which will be `isRequired`. In this entry the user can add a new group or choose from a list of groups present in his project.
- When user doesn't select any group name, generate a group name for the route using [haikunator](https://github.com/Atrox/haikunatorjs). To check the overlapping probability of the algorithm for 1 million values.
- Send the payload to be handled by the backend and persisting in the local-config.

### Backend
- local-config doesn't only contain a routes array but a routes map whose hashed key will be the group the route belongs to.
- the modification of the array into map would mean that the monitoring submodule of bench routes will be modified to include the routes from this map.
- The group name would essentially be able to fetch consistently a group of routes under it without any lock or errors. Better to write some tests for that.

The UI would be beautifully designed :sparkles:

**Comment by Harkishen-Singh** [comment]
A clean UI is prefered by all. So, this should not be an optional thing from the `quick-input` screen's perspective. What we should rather do is that in the Monitoring screen, give a checkbox in the top as `Group`. If checked, the routes will be grouped by domain names (via backend). This by no means requires a change in the quick-input screen. 

**Approach**

The approach remains pretty straight forward. While sending the response for `Monitoring` screen, implement a `map[string][]struct{<matrix_instances>}` where the key corresponds to `domain name` and the value is a `Matrix` instance which would be modified a bit according to the needs.

When the `group` is checked in the UI, the Matrix would be made to run through a loop where the `URL` field of `Matrix` will be passed through `filters.HTTPPingFilterValue()` and if that key does not exist in the map, then add the key to the map and initialize the value of the map as that matrix item. This will be a linear time complexity, hence consuming almost the same time when we are not grouping (since in the non-grouping case, we still go through a loop). 

The above method also makes it clear that there is no requirement of a tag since even after having a tag, the time complexity will again same to what I mentioned. So, better not go by the tag way.

**Comment by aquibbaig** [comment]
@Harkishen-Singh I do not think it will be a good idea to checkbox here in terms of UI perspective. People most necessary would want to do the seggregation once and for all when they input a route and nobody will spend time checking for `groups` later on. The point of having this is basically having a once and for all manual input and rest is automation.
